src/redux/slices/onboarding.ts

```diff
@@ -0,0 +1,84 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { RootState } from '@/redux/RootState';
+import SliceName from '@/redux/types/SliceName';
+import OnboardingGroup from '@/types/OnboardingGroup';
+
+type ActiveStepPayload = { group: OnboardingGroup; index: number };
+
+export type OnboardingState = {
+  isChecklistVisible: boolean;
+  checklistDismissals: number;
+  lastChecklistDismissal: Date | null;
+  activeStep: ActiveStepPayload;
+  groups: Partial<
+    Record<
+      OnboardingGroup,
+      {
+        completedSteps: number[];
+      }
+    >
+  >;
+};
+
+const initialState: OnboardingState = {
+  isChecklistVisible: true,
+  checklistDismissals: 0,
+  lastChecklistDismissal: null,
+  activeStep: {
```

Comment:
nit: `activeStep` should be renamed to `activeGroup` or `activeGroupAndStep` since it's not only holding the current step but also the current group.

Reply:
`activeStep` is the combination of `group` + `index`. A step is unique throughout the onboarding experience. Imo the current name is fine